### PR TITLE
Ability to Preview Crates

### PR DESCRIPTION
**Pull request by Heisenburger69**
Please make sure your issue complies with these guidelines:
- * [x] Use same formatting
- * [x] Changes must have been tested on PMMP.
- * [x] Unless it is a minor code modification, you must use an IDE.
- * [x] Have a detailed title (i.e "Fix Crate items don't appear in inventory")

### What does the PR change?

This PR aims to add the ability to preview the items within a crate. The RouletteTask has been modified to loop through all possible drops endlessly without giving the player an item, until the player closes the menu.  

### Testing Environment

* PocketMine-MP: 3.11.5
* PHP: 7.2
* Server OS: Windows 10

### Extra Information
The InvMenu virion is used.

### CODE REVIEW

File: src/DaPigGuy/PiggyCrates/tasks/RouletteTask.php [hunk 1]
```diff
@@ -26,14 +30,36 @@ class RouletteTask extends Task
     /** @var CrateItem[] */
     private $lastRewards = [];
 
-    public function __construct(CrateTile $tile)
+    /** @var bool */
+    private $preview = false;
+    /**
```
Comment (DaPigGuy) [inline src/DaPigGuy/PiggyCrates/tasks/RouletteTask.php]: Formatting for this and the PHPDoc below 😅

File: src/DaPigGuy/PiggyCrates/tiles/CrateTile.php [hunk 2]
```diff
@@ -101,6 +101,13 @@ public function closeCrate(): void
         $this->currentPlayer = null;
     }
 
+    public function previewCrate(Player $player): void
+    {
```
Comment (DaPigGuy) [inline src/DaPigGuy/PiggyCrates/tiles/CrateTile.php]: Did you really need to do the preview in the roulette task? :l

Comment (Heisenburger69) [reply src/DaPigGuy/PiggyCrates/tiles/CrateTile.php]: Didn't see a point of a new Task when it's pretty much the same code

**Comment by Aericio** [comment]
@DaPigGuy change your review smh

**Pull request merged**